strawberryfields/apps/sample.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+r"""
+Sampling functions
+==================
+
+**Module name:** :mod:`strawberryfields.future.apps.sample`
+
+.. currentmodule:: strawberryfields.future.apps.sample
+
+This module provides functionality for generating samples from both a quantum device and the
+uniform distribution.
+
+The :func:`quantum_sampler` function allows users to simulate Gaussian boson sampling (GBS) by
+choosing a symmetric input matrix to sample from. For a symmetric :math:`N \times N` input
+matrix :math:`A`, corresponding to an undirected graph, an :math:`N`-mode GBS device with threshold
+detectors generates samples that are binary strings of length ``N``. Various problems can be
+encoded in the matrix :math:`A` so that the output samples are informative and can be used as a
+form of randomness to improve solvers, as outlined in Refs. :cite:`arrazola2018using` and
+:cite:`arrazola2018quantum` and also shown in our :ref:`tutorial_dense_subgraph` tutorial.
+
+On the other hand, the :func:`uniform_sampler` function allows users to generate samples where a
+subset of modes are selected using the uniform distribution.
+
+Summary
+-------
+
+.. autosummary::
+    QUANTUM_BACKENDS
+    BACKEND_DEFAULTS
+    quantum_sampler
+    uniform_sampler
+    random_seed
+
+Code details
+------------
+"""
+from typing import Optional
+import numpy as np
+import strawberryfields as sf
+from .graph import utils
+
+QUANTUM_BACKENDS = ("gaussian",)
+"""tuple[str]: Available quantum backends for sampling."""
+
+BACKEND_DEFAULTS = {
+    "remote": False,
+    "backend": "gaussian",
+    "threshold": True,
+    "postselect": 0,
+}
+"""Dict[str, Any]: Dictionary to specify default parameters of options in backend sampling for
+:func:`quantum_sampler`.
+"""
+
+
+def _valid_backend(backend: str) -> bool:
+    r"""Performs a validation check that the requested backend can be run based upon the
+    ``QUANTUM_BACKENDS`` tuple.
+
+    Args:
+        backend (str): requested backend for sampling
+
+    Returns:
+        bool: returns only ``True`` if a valid backend is selected
+    """
+    return backend in QUANTUM_BACKENDS
```

Comment:
Since this function only performs a (relatively simple) one line operation, it would probably make the code clearer/easier to follow to remove this function, and have the line `backend in QUANTUM_BACKENDS` used directly where it is called

Reply:
Removed!